### PR TITLE
TRUNK-4862 Resolve syntax error in errorhandler.jsp

### PR DESCRIPTION
**Pull request by bgeVam**
see https://issues.openmrs.org/browse/TRUNK-4862

### CODE REVIEW

File: webapp/src/main/webapp/errorhandler.jsp [hunk 1]
```diff
@@ -50,5 +50,5 @@ else {
 	out.println("-->");
 }
 org.openmrs.api.context.Context.closeSession();
-
```
Comment (lluismf) [inline webapp/src/main/webapp/errorhandler.jsp]: I'd indent lines 23-53 to make it more readable

Comment (bgeVam) [inline webapp/src/main/webapp/errorhandler.jsp]: PR Updated,
i ran ctrl+shift+f to format the jsp

**Comment by teleivo** [comment]
@lluismf or @dkayiwa I have seen this travis CI error for jdk6 in the radiology module as well

see https://github.com/travis-ci/travis-ci/issues/5227

I added following workaround in .travis.yml
before_install:
  - cat /etc/hosts # optionally check the content *before*
  - sudo hostname "$(hostname | cut -c1-63)"
  - sed -e "s/^\\(127\\.0\\.0\\.1.*\\)/\\1 $(hostname | cut -c1-63)/" /etc/hosts | sudo tee /etc/hosts
  - cat /etc/hosts # optionally check the content *after*

also I had to change running travis CI in containerized version since I hit their 3GB RAM limit (I guess due to component tests). Here is my PR on the radiology module https://github.com/openmrs/openmrs-module-radiology/pull/247/files

File: webapp/src/main/webapp/errorhandler.jsp [hunk 2]
```diff
@@ -1,54 +1,53 @@
-<%@ page isErrorPage="true" import="java.io.*" %>
+<%@ page isErrorPage="true" import="java.io.*"%>
```
Comment (wluyima) [inline webapp/src/main/webapp/errorhandler.jsp]: It's actually good practice and cleaner to leave this space at the end, therefore it was intentional and I think you should undo this

File: webapp/src/main/webapp/errorhandler.jsp [hunk 3]
```diff
@@ -1,54 +1,53 @@
...
 
-<%@ taglib uri="/openmrs" prefix="openmrs" %>
+<%@ taglib uri="/openmrs" prefix="openmrs"%>
```
Comment (wluyima) [inline webapp/src/main/webapp/errorhandler.jsp]: Same here

File: webapp/src/main/webapp/errorhandler.jsp [hunk 4]
```diff
@@ -1,54 +1,53 @@
...
-	exception = exception.getCause();
+	if (exception.getCause() != null
+			&& (ContextAuthenticationException.class.equals(exception.getCause().getClass())
+					|| APIAuthenticationException.class.equals(exception.getCause().getClass()))) {
+		//convert it back to the actual exception that was thrown
+		exception = exception.getCause();
 %>
 
-<%@ include file="/WEB-INF/view/authorizationHandlerInclude.jsp" %>
+<%@ include file="/WEB-INF/view/authorizationHandlerInclude.jsp"%>
```
Comment (wluyima) [inline webapp/src/main/webapp/errorhandler.jsp]: And here

File: webapp/src/main/webapp/errorhandler.jsp (hunk 2 above)
Comment (wluyima) [inline webapp/src/main/webapp/errorhandler.jsp]: I already merged the code, never mind

Comment (lluismf) [inline webapp/src/main/webapp/errorhandler.jsp]: Wildcard imports are discouraged (in JSP also)

Comment (wluyima) [inline webapp/src/main/webapp/errorhandler.jsp]: I agree

Comment (wluyima) [inline webapp/src/main/webapp/errorhandler.jsp]: In fact we discourage them throughout our code base including classes

**Pull request merged**